sympy/core/sympify.py

```diff
@@ -99,7 +99,7 @@ def _convert_numpy_types(a, **sympify_args):
 
 
 def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
-        evaluate=None):
+        evaluate=None, controlled=True):
```

Comment:
Is `True` by default the right choice? Does this come with some performance hit to the library while it is not needed for private use? Perhaps a global parameter to switch the default behaviour might be useful.

Reply:
Same answer, this is just here temporarily for experimentation, to force all unit tests to go through the new evaluator.

Reply:
Why "controlled" instead of "safe" like at https://github.com/sympy/sympy/pull/12524?

Reply:
I just didn't want to use "safe" until I was 100% sure. I think I'd feel better about using that word with an implementation such as discussed [here](https://github.com/sympy/sympy/issues/10805#issuecomment-1047956743), where the whitelist of callables would be explicit, and would list only functions each of which had been thoroughly audited.

Reply:
It's true that "safe" may imply a stronger guarantee than we can actually provide. But I also feel like it's a much more self descriptive name for the argument. It's also not easy to change keyword argument names in the future so we should try to pick a good one.